### DISCUSSION TITLE
Capacitor Support

### DISCUSSION THREAD

**Discussion by Excel1** [keycloak/keycloak#30657] [Ideas]
I know that it is possible to add further adapters. Besides Cordova, Capacitor/ionic is one of the most popular app platforms in the Playstore. I think it makes sense to officially support capacitor like cordova and bring a capacitor-native and capacitor adapter. Currently, the implementation is private due to many private variables, as can be seen in other discussions, making a reasonable implementation very complex or switching to other libraries. There are good libraries for Angular and many other frameworks, but there is also no good library for VueJs, for example, which increases the effort many times over

**Comment by jonkoops** [comment]
We do not intend to further write or maintain adapters for these platforms as we simply do not have the capacity to do so. Keycloak uses standardized protocols such as OpenID Connect and OAuth, for which many client implementations are available. My recommendation would be to use whatever client is popular and supported for the platform you are interested in developing for.
